validations/src/validations.rs

```diff
@@ -1326,6 +1326,7 @@ pub fn validate_block_transactions(
     block_number: u32,
     consensus_constants: &ConsensusConstants,
     active_wips: &ActiveWips,
+    priorities: &mut Priorities,
```

Comment:
Yes, that's the thing. No easy remediation.

Reply:
Hmm I got an idea on how to improve this. Let me draft something and will push a commit soon.

Reply:
Ok, I have refactored these validation functions so instead of receiving a mutable reference to a `Priorities` object, they receive an implementer of `trait Visitor`.

In this way, we can extend the functionality of validation functions at any time without changing their internal business logic, breaking any tests, nor passing any further arguments, as these visitors act as very convenient "plug-ins" that can visit data in an immutable fashion. They can even be combined together (see `impl Visitor for DualVisitor`) or be created from a simple closure (`impl Visitor for GenericVisitor`).

Reply:
I was expecting a less generic visitor, such as

```rust
trait BlockTransactionsVisitor {
    fn visit_vtt(&mut self, vtt: ...) {}
    fn visit_drt(&mut self, vtt: ...) {}
}
```

But I guess it's fine, I don't expect this feature to have many use cases aside from this one. Maybe we could reimplement some of the existing validations as visitors, such as a visitor that checks that the total weight does not exceed the block limit, or a visitor that calculates the merkle root, but I don't see any benefits from doing that.

Reply:
`signatures_to_verify` looks like a candidate for refactoring, as from a high level it apparently follows the same pattern of being passed as a `&mut`.